util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/rule/SearchWorkCall.java

```diff
@@ -44,13 +60,35 @@
 				.as( "Search work on indexes " + indexNames + " did not match: " )
 				.matches( work );
 		/*
-		 * Checking the result type in the work assertion above guarantees
-		 * that the hit collector has the correct type. We can cast safely.
+		 * Checking the result type in the work assertion above guarantees that the hit collector has the correct type.
+		 * We can cast safely.
 		 */
 		long totalHitCount = behavior.getTotalHitCount();
-		@SuppressWarnings({ "unchecked", "rawtypes" })
-		List<U> hits = (List<U>) ((StubHitExtractor) actualCall.hitExtractor).extract( behavior.getRawHits() );
-		return new SearchResultImpl<>( totalHitCount, hits );
+
+		return new SearchResultImpl<>( totalHitCount, getResults( actualCall.convertContext,
+				actualCall.projectionHitMapper, actualCall.rootProjection, behavior.getRawHits() ) );
+	}
+
+	private static <U> List<U> getResults(FromIndexFieldValueConvertContext actualConvertContext,
```

Comment:
Keeping the `StubHitExtractor` class and encapsulating the `convertContext`, `projectionHitMapper` and `rootProjection` in it would have spared you this code, you would just have delegated to the extractor. But I suppose it's just as well.

Reply:
So the idea is that I would like to have the Stub part as consistent with the rest as possible. It's already quite different tbh and it's a bit hard to make the changes here after doing the changes in the backends.